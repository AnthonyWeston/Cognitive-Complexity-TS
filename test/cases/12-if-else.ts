function f() {
    if (true) {
    } else if (true) {
    } else {
    }
}

function g() {
    // depth 0
    if (true) { // +1
        // depth 1
        while (true); // +1 +1
    } else if (true) { // +1
        // depth 1
        while (true); // +1 +1
    } else { // +1
        // depth 1
        while (true); // +1 +1
    }
}

function h() {
    if (true) {
    }
}


function i() {
    if (true) {
    } else if (true) {
    }
}

function j() {
    if (true) {
    } else {

    }
}
