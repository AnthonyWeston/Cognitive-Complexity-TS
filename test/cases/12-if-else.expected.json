{
    "12-if-else.ts": {
        "score": 17,
        "inner": [
            {
                "name": "f",
                "score": 3,
                "line": 1,
                "column": 1
            },
            {
                "name": "g",
                "score": 9,
                "line": 8,
                "column": 1
            },
            {
                "name": "h",
                "score": 1,
                "line": 22,
                "column": 1
            },
            {
                "name": "i",
                "score": 2,
                "line": 28,
                "column": 1
            },
            {
                "name": "j",
                "score": 2,
                "line": 34,
                "column": 1
            }
        ]
    }
}